scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl

```diff
@@ -99,9 +93,22 @@ workflow GvsExtractCohortFromSampleNames {
   }
 
   File cohort_sample_names_file = select_first([write_array_task.output_file, cohort_sample_names])
+  Int effective_sample_count = length(read_lines(cohort_sample_names_file))
+
+  Int effective_scatter_count = if defined(extract_scatter_count_override) then select_first([extract_scatter_count_override])
+                                else if is_wgs then
+                                     if effective_sample_count < 5000 then 1 # This results in 1 VCF per chromosome.
+                                     else if effective_sample_count < 20000 then 2000 # Stroke Anderson
+                                          else if effective_sample_count < 50000 then 10000
+                                               else 20000
+                                     else
+                                     if effective_sample_count < 5000 then 1 # This results in 1 VCF per chromosome.
+                                     else if effective_sample_count < 20000 then 1000
+                                          else if effective_sample_count < 50000 then 2500
+                                               else 7500
```

Comment:
Ridiculously wide scatters during cohort extract have been super annoying for a long time, and in this work they were actually breaking P&S runs when trying to merge all thousands of VCFs that were >99% header lines. Simple fixes here largely copy/pasted from the regular extrace workflow.

Reply:
Smart -- we know VCF creation is very much NOT linear, so building our step function this way makes more sense